src/ctx.cpp

```diff
@@ -3765,15 +3792,16 @@ llvm::Value *FunctionEmitContext::addVaryingOffsetsIfNeeded(llvm::Value *ptr, co
     bool is32bits = g->target->is32Bit() || g->opt.force32BitAddressing;
     llvm::Value *varyingOffsets = ProgramIndexVector(is32bits);
 
-    llvm::Value *offset = BinaryOperator(llvm::Instruction::Mul, unifSize, varyingOffsets);
+    llvm::Value *offset = BinaryOperator(llvm::Instruction::Mul, unifSize, varyingOffsets,
+                                         baseType->IsSignedType() ? WrapSemantics::NSW : WrapSemantics::None);
```

Comment:
How the "pointed to" type affects the NSW bit of address computation? Am I missing something here?

Reply:
Misread the type of unifSize. Fixed.